pkg/controller/plan/scheduler/vsphere/scheduler.go

```diff
@@ -0,0 +1,117 @@
+package vsphere
+
+import (
+	"context"
+	liberr "github.com/konveyor/controller/pkg/error"
+	api "github.com/konveyor/forklift-controller/pkg/apis/forklift/v1alpha1"
+	"github.com/konveyor/forklift-controller/pkg/apis/forklift/v1alpha1/plan"
+	plancontext "github.com/konveyor/forklift-controller/pkg/controller/plan/context"
+	"github.com/konveyor/forklift-controller/pkg/controller/provider/web/vsphere"
+)
+
+//
+// Scheduler for migrations from ESX hosts.
+type Scheduler struct {
+	*plancontext.Context
+	MaxInFlight int
+	inflight    map[string]int
+	pending     map[string][]*pendingVM
+}
+
+type pendingVM struct {
+	status *plan.VMStatus
+	cost   int
+}
+
+//
+// Return the next VM to migrate.
+func (r *Scheduler) Next() (vm *plan.VMStatus, err error) {
```

Comment:
Although there are examples of this in the MT code, I would recommend against using `nil` as = `has next`.
Recommend and explicit: 
```
func (r *Scheduler) Next() (vm *plan.VMStatus, hasNext bool, err error) 
```

Or return a list of VMs that can be started.  The list could be empty.
```
```func (r *Scheduler) Next() (vm []*plan.VMStatus, err error) {